valohai_cli/utils/ssh.py

```diff
@@ -0,0 +1,144 @@
+import json
+import os
+import pathlib
+import shutil
+import subprocess
+from dataclasses import dataclass
+from typing import Generator, Iterable, Iterator, Optional, Tuple
+
+import click
+
+from valohai_cli.api import request
+from valohai_cli.ctx import get_project
+from valohai_cli.settings import get_settings_file_name
+from valohai_cli.utils import call_with_retry
+from valohai_cli.utils.cli_utils import prompt_from_list
+
+
+@dataclass(frozen=True)
+class Keypair:
+    public_path: pathlib.Path
+    private_path: pathlib.Path
+
+
+def make_ssh_connection_to_container(address: str, port: int, private_key_path: str) -> None:
```

Comment:
I don't think there's much value in saying `_to_container` here?
```suggestion
def make_ssh_connection(address: str, port: int, private_key_path: str) -> None:
```